modules/Emails/EmailsController.php

```diff
@@ -639,4 +673,78 @@ protected function setAfterImport($importedEmailId, $request)
 
         return $emails;
     }
+
+    /**
+     * @param User $requestedUser
+     * @param InboundEmail $requestedInboundEmail
+     * @param Email $requestedEmail
+     * @return bool false if user doesn't have access
+     */
+    protected function userIsAllowedToSendEmail($requestedUser, $requestedInboundEmail, $requestedEmail)
+    {
+        $hasAccess = false;
+
+        // Check that user is allowed to use inbound email account
+        $hasAccessToInboundEmailAccount = false;
+        $usersInboundEmailAccounts = $requestedInboundEmail->retrieveAllByGroupIdWithGroupAccounts($requestedUser->id);
+        foreach ($usersInboundEmailAccounts as $inboundEmailId => $userInboundEmail) {
+            if ($userInboundEmail->id === $requestedInboundEmail->id) {
+                $hasAccessToInboundEmailAccount = true;
+                break;
+            }
+        }
+
+        $inboundEmailStoredOptions = $requestedInboundEmail->getStoredOptions();
+
+        // if group email account check that user is allowed to use email account
+        if ($requestedInboundEmail->isGroupEmailAccount()) {
+            if ($inboundEmailStoredOptions['allow_outbound_group_usage'] === true) {
+                $hasAccessToInboundEmailAccount = true;
+            } else {
+                $hasAccessToInboundEmailAccount = false;
+            }
+        }
+
+        // Check that the from address is the same as the inbound email account
+        $isFromAddressTheSame = false;
+        if ($inboundEmailStoredOptions['from_addr'] === $requestedEmail->from_addr) {
+            $isFromAddressTheSame = true;
+        }
+        // check if user is using the system account, as the email address for the system account
+        // is likely to be different
+        $outboundEmailAccount = new OutboundEmail();
+        if(empty($inboundEmailStoredOptions['outbound_email'])) {
+            $outboundEmailAccount->getUserMailerSettings();
+        } else {
+            $outboundEmailAccount->retrieve($inboundEmailStoredOptions['outbound_email']);
+        }
+
+        $isAllowedToUseOutboundEmail = false;
+        if ($outboundEmailAccount->type === 'system') {
```

Comment:
I see. Then we're changing the current behavior of defaulting to the system `outbound_email` account, when the wanted `outbound_email` does not exist. I actually have an account in this situation on my dev instance (it wasn't done by hand..).

In this case it may not make sense to blindly destroy the user's session, as the user may tried to use a non-existing outbound account associated with his inbound account, as opposed to trying to use an account he does not have access to.